File: maxhanna.client/src/services/datacontracts/meta/meta-hero.ts
import { User } from "../user/user"; 
import { Vector2 } from "./vector2";
import { MetaBot } from "./meta-bot";

export class MetaHero {
  id: number;  
  name?: string;
  position: Vector2; 
  speed: number;
  map: string;
  color?: string;
  metabots: MetaBot[];

  constructor(id: number, name: string, position: Vector2, speed: number, map: string, metabots: MetaBot[], color?: string) {
    this.id = id; 
    this.name = name;
    this.position = position; 
    this.speed = speed;
    this.map = map;
    this.color = color;
    this.metabots = metabots;
  } 
}

File: maxhanna.Server/Controllers/DataContracts/Meta/MetaHero.cs
using maxhanna.Server.Controllers.DataContracts.Users;

namespace maxhanna.Server.Controllers.DataContracts.Meta
{
    public class MetaHero
    { 
        public string? Name { get; set; }
        public int Id { get; set; }
        public Vector2 Position { get; set; } = new Vector2(0,0);
        public List<MetaBot>? Metabots { get; set; }
        public int Speed { get; set; }
        public string Map { get; set; } = "";
        public string Color { get; set; } = "";
        public int? Mask { get; set; } = null;
    }
}
